packages/remark-mdx/index.js

```diff
@@ -102,19 +101,12 @@ function tokenizeEsSyntax(eat, value) {
   const index = value.indexOf(EMPTY_NEWLINE)
   const subvalue = index !== -1 ? value.slice(0, index) : value
 
-  if (isExport(subvalue)) {
-    return eat(subvalue)({
-      type: 'export',
-      default: isExportDefault(subvalue),
-      value: subvalue
-    })
-  }
+  if (isExport(subvalue) || isImport(subvalue)) {
+    const [first, ...rest] = extractImportsAndExports(subvalue)
+
+    eat(subvalue)(first)
```

Comment:
This way loses position information, which could be a problem for tooling in the remark/mdx ecosystems (not for MDX -> JSX). Now `first` gets the positional info for the whole eaten value, and `rest` gets the position at the end of `value` as it’s start and end positions.

You could use `eat.now()`  first, to get the position we’re at, and then I’d believe babel in `extractImportsAndExports` has access to the positions of the import/export nodes, so you could patch the positions there.
Then finally, you could `eat(subvalue)` to mark the text as parsed, but that not use the returned function, as the nodes already have position info attached?

Reply:
Ah, nice, yeah I figured I wasn't doing something right there. Will make that fix, thx!

Reply:
It could potentially be a bit hard to figure out the positions, e.g., if the block starts (for remark) at 5:3 (L5C3), you’ll have to do some maths to adjust the Babel nodes 😬

P.S. Babel uses 0-indexed columns, unist uses 1-indexed!